karapuzio/assignment3/WordSearch.java

```diff
@@ -0,0 +1,141 @@
+package edu.codeU.assignment3;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * The class to find all words in the grid.
+ */
+public class WordSearch {
+    /**
+     * The number of rows in grid.
+     */
+    private int rows;
+    /**
+     * The number of columns in grid;
+     */
+    private int columns;
+    /**
+     * The grid in which we look words.
+     */
+    private char[][] grid;
+    /**
+     * The List of Words in dictionary.
+     */
+    private List<String> dict;
+    /**
+     * Flag array that use in dfs to check existing of current cell of grid at current string.
+     */
+    private boolean flag[][];
+
+    public WordSearch(int rows, int columns, char[][] grid, List<String> dict) {
+        this.rows = rows;
+        this.columns = columns;
+        this.grid = grid;
+        this.dict = dict;
+        flag = new boolean[rows][columns];
+    }
+
+    /**
+     * We find x, y - coordinates in grid of all neighbors.
+     * @param x is the current row.
+     * @param y is the current column.
+     * @param neighborX the x - coordinates of all neighbors.
+     * @param neighborY the y - coordinates of all neighbors.
+     */
+    private void getAllNeighbors(int x, int y, List<Integer> neighborX, List<Integer> neighborY){
+        int SIZE_DELTAS = 3;
+        int[] deltaX = {-1, 0, 1};
```

Comment:
These two arrays of deltas can be static and final and defined outside of this method, then you will save on creating them every time you call getAllNeighbors.
Also you can just use deltaX.size instead of additional variable SIZE_DELTAS :)